dpnp/dpnp_algo/dpnp_algo_indexing.pxi

```diff
@@ -203,39 +162,6 @@ cpdef utils.dpnp_descriptor dpnp_diagonal(dpnp_descriptor x1, offset=0):
     return result
 
 
-cpdef dpnp_fill_diagonal(dpnp_descriptor x1, val):
-    x1_obj = x1.get_array()
-
-    cdef shape_type_c x1_shape = x1.shape
-    cdef utils.dpnp_descriptor val_arr = utils_py.create_output_descriptor_py((1,),
-                                                                              x1.dtype,
-                                                                              None,
-                                                                              device=x1_obj.sycl_device,
-                                                                              usm_type=x1_obj.usm_type,
-                                                                              sycl_queue=x1_obj.sycl_queue)
-
-    val_arr.get_pyobj()[0] = val
-
-    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(x1.dtype)
-
-    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_FILL_DIAGONAL_EXT, param1_type, param1_type)
-
-    cdef c_dpctl.SyclQueue q = <c_dpctl.SyclQueue> x1_obj.sycl_queue
-    cdef c_dpctl.DPCTLSyclQueueRef q_ref = q.get_queue_ref()
-
-    cdef custom_indexing_2in_func_ptr_t func = <custom_indexing_2in_func_ptr_t > kernel_data.ptr
```

Comment:
please remove definition of `custom_indexing_2in_func_ptr_t`, since never used now